apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/PluginFinder.java

```diff
@@ -95,13 +95,13 @@ public boolean matches(NamedElement target) {
                 return nameMatchDefine.containsKey(target.getActualName());
             }
         };
-        judge = judge.and(not(isInterface()));
         for (AbstractClassEnhancePluginDefine define : signatureMatchDefine) {
             ClassMatch match = define.enhanceClass();
             if (match instanceof IndirectMatch) {
                 judge = judge.or(((IndirectMatch) match).buildJunction());
             }
         }
+        judge = judge.and(not(isInterface()));
```

Comment:
```suggestion
        // Filter out all matchers returns to exclude pure interface types.
        judge = not(isInterface()).and(judge);
```

I think this should be the way you propose. The result will be the same as your wrote, but the logic is more clear.

`not an interface` and `all other pre-built checker`.